porttime/pthaiku.cpp

```diff
@@ -0,0 +1,88 @@
+// pthaiku.cpp - portable timer implementation for Haiku
```

Comment:
Is there a (good/necessary) reason to write this in C++? Portmidi is in C and uses the C runtime, so adding C++ (I think) means every application, even those in C, would require the C++ runtime library.

Reply:
PortTime would not strictly be necessary because the POSIX/Linux version is nearly compatible. The only thing that is missing is `CLOCK_MONOTONIC_RAW` which would have to be replaced with `CLOCK_MONOTONIC`. I don't know what exactly would be the difference between those two.

The Midi kit however, is only available in C++, so the actual PortMidi implementation needs to use C++.